packages/opentelemetry-context-async-hooks/src/AsyncHooksContextManager.ts

```diff
@@ -52,72 +39,36 @@ const ADD_LISTENER_METHODS = [
 
 export class AsyncHooksContextManager implements ContextManager {
   private _asyncHook: asyncHooks.AsyncHook;
-  private _contextRefs: Map<number, Reference<Context> | undefined> = new Map();
+  private _contexts: Map<number, Context | undefined> = new Map();
+  private _stack: Array<Context | undefined> = [];
```

Comment:
If you have a single global stack, isn't it not possible for two concurrent requests to have different contexts?

Reply:
It not possible because NodeJS is singlethreaded, even if they looks like concurrent they are not. Plus `asynchooks` allow us to restore the context just before we run the userland code and remove it just after.

Reply:
I've added a test case where i tried to represent two "concurrent" operation.

Reply:
I saw that. I tested some locally too and couldn't break it so I think I'm satisfied.